client/components/seo/twitter-preview/index.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { PropTypes } from 'react';
+import PureRenderMixin from 'react-pure-render/mixin';
+import classNames from 'classnames';
+import crypto from 'crypto';
+
+const baseDomain = url =>
+	url
+		.replace( /^[^/]+[/]*/, '' ) // strip leading protocol
+		.replace( /\/.*$/, '' ); // strip everything after the domain
+
+const blavatarUrl = slug => {
+	const slugHash = crypto
+		.createHash( 'md5' )
+		.update( slug )
+		.digest( 'hex' );
+
+	return `https://secure.gravatar.com/blavatar/${ slugHash }?s=512`;
+};
+
+export const TwitterPreview = React.createClass( {
+	mixins: [ PureRenderMixin ],
+
+	render() {
+		const {
+			url,
+			title,
+			type,
+			description,
+			image,
+			slug
+		} = this.props;
+
+		const previewClasses = classNames( 'twitter-card-preview', {
+			[ `twitter-card-preview__${ type }` ]: true
+		} );
+
+		return (
+			<div className={ previewClasses }>
+				<div className="twitter-card-preview__image">
+					<img className="" src={ image || blavatarUrl( slug ) } />
+				</div>
+				<div className="twitter-card-preview__body">
+					<div className="twitter-card-preview__title">
+						{ title || '' }
```

Comment:
These are fine to leave without the default blank. The reason I put them in above is because the truncation functions needed a string to work and I didn't want to change them. Here `{ title }` would be fine because a blank string, `undefined`, or `null` values for `title`, `description`, etc… just show up blank in React.